### PR TITLE
mail-filter/sqlgrey - systemd service improvements and hardening

### PR DESCRIPTION
**Pull request by candrews**
https://bugs.gentoo.org/show_bug.cgi?id=587596

### CODE REVIEW

**Comment by monsieurp** [comment]
Thanks @candrews for your work and your PRs. Please slow down the pace at which you file PRs to allow us to review the backlog we've accumulated so far. Cheers.

**Comment by mgorny** [comment]
@candrews, thanks a lot for all what you're doing to improve systemd support in Gentoo. Would you be interested in joining our systemd team by any chance? Due to technical limitations we can't add you to the wiki listing but I can add you to the systemd@ mail alias, so you'll receive all the relevant mail ;-).

**Comment by candrews** [comment]
@mgorny I'd be happy to join the team, thanks for the offer!

**Comment by mgorny** [comment]
@candrews, I'm sorry that this took this long but I've finally added you to the alias. So don't be surprised of the new systemd spam you're going to get now ;-).

**Comment by candrews** [comment]
@mgorny thanks!

I realized that this PR should include a revbump so I updated it to include that.

Can you please review and (hopefully) merge this now?

**Comment by mgorny** [comment]
@gentoo/systemd

**Comment by monsieurp** [comment]
Hi! Is someone from the systemd team looking into merging this PR? @mgorny?

File: mail-filter/sqlgrey/files/sqlgrey.service [hunk 1]
```diff
@@ -3,7 +3,19 @@ Description=SQLgrey Postfix Grey-listing Policy service
 After=network.target
 
 [Service]
+User=sqlgrey
+Group=sqlgrey
 ExecStart=/usr/sbin/sqlgrey
+PermissionsStartOnly=true
+ExecStartPre=/usr/bin/touch /run/sqlgrey.pid && /bin/chown sqlgrey:sqlgrey /run/sqlgrey.pid
```
Comment (mgorny) [inline mail-filter/sqlgrey/files/sqlgrey.service]: I don't like this. Any reason not to use tmpfiles.d? Any reason to combine it with `PIDFile` below?

Comment (candrews) [inline mail-filter/sqlgrey/files/sqlgrey.service]: I rethought this and I don't think `PIDFile` needs to be specified, so /run/sqlgrey.pid also need not be created. I've updated the PR accordingly.

**Comment by mgorny** [comment]
@candrews, could you confirm that you've tested the final version?

**Comment by mgorny** [comment]
In any case, this is systemd team-approved. When @candrews or anyone else confirms that the final version works, feel free to merge it.

**Comment by candrews** [comment]
The changes in this PR work fine - I've been using them and I'm getting email handled by sqlgrey without issue.

**Pull request closed without merging**